android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.filament.utils
+
+import android.view.MotionEvent
+import android.view.Surface
+import android.view.SurfaceView
+import com.google.android.filament.*
+import com.google.android.filament.android.UiHelper
+import com.google.android.filament.gltfio.*
+import java.nio.Buffer
+
+/**
+ * Helper class for rendering glTF models with an orbit controller.
+ *
+ * `ModelViewer` owns a Filament engine, renderer, swapchain, view, and scene. It allows clients
+ * to access these objects via read-only properties. By design, the viewer can display only one
+ * glTF scene at a time. The scene can be scaled and translated into the viewing frustum by
+ * calling [transformToUnitCube]. All ECS entities can be accessed via the [asset] property.
+ *
+ * For GLB files, clients can call [loadModelGlb] and pass in a [Buffer] with the contents of the
+ * GLB file. For glTF files, clients can call [loadModelGltf] and pass in a [Buffer] with the JSON
+ * contents, as well as a callback for loading external resources.
+ *
+ * `ModelViewer` reduces much of the boilerplate required for simple Filament applications, but
+ * clients still have the responsibility of adding an [IndirectLight] and [Skybox] to the scene.
+ * Additionally, clients should:
+ *
+ * 1. Call [onTouchEvent] from a touch handler.
+ * 2. Call [render] and [Animator.applyAnimation] from a `Choreographer` frame callback.
+ * 3. Call [detach] when the model viewer is no longer needed.
+ *
+ * See `sample-gltf-viewer` for a usage example.
+ */
+class ModelViewer(val surfaceView: SurfaceView) {
```

Comment:
Could you make it work with either a SurfaceView or TextureView? Ideally it would work with any Surface.

Reply:
Done, took an approach similar to UiHelper by providing two overloads.